src/Common/tests/System/Net/Http/Http2LoopbackServer.cs

```diff
@@ -374,13 +374,30 @@ private static (int bytesConsumed, string value) DecodeString(ReadOnlySpan<byte>
             }
         }
 
-        private static int EncodeString(string value, Span<byte> headerBlock)
+        private static int EncodeString(string value, Span<byte> headerBlock, bool huffmanCode)
         {
-            int bytesGenerated = EncodeInteger(value.Length, 0, 0x80, headerBlock);
+            byte[] data = Encoding.ASCII.GetBytes(value);
+            byte prefix;
 
-            bytesGenerated += Encoding.ASCII.GetBytes(value.AsSpan(), headerBlock.Slice(bytesGenerated));
+            if (!huffmanCode)
+            {
+                prefix = 0;
+            }
+            else
+            {
+                int len = HuffmanEncoder.GetEncodedLength(data);
 
-            return bytesGenerated;
+                byte[] huffmanData = new byte[len];
+                HuffmanEncoder.Encode(data, huffmanData);
+
+                data = huffmanData;
+                prefix = 0x80;
+            }
+
+            int bytesGenerated = EncodeInteger(data.Length, prefix, 0x80, headerBlock);
+            data.AsSpan().CopyTo(headerBlock.Slice(bytesGenerated));
+
+            return bytesGenerated + data.Length;
```

Comment:
Nit: The general pattern here is to always return "bytesGenerated" at the end of the function. So it would be slightly clearer to do something like:

```csharp
    data.AsSpan().CopyTo(headerBlock.Slice(bytesGenerated));
    bytesGenerated += data.Length;

    return bytesGenerated;
```